pkg/frontend/outbound_type.go

```diff
@@ -0,0 +1,28 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/feature"
+)
+
+func determineOutboundType(ctx context.Context, doc *api.OpenShiftClusterDocument, subscription *api.SubscriptionDocument) error {
+	var err error
+	// Honor the value of OutboundType if it was passed in via the API
+	if doc.OpenShiftCluster.Properties.NetworkProfile.OutboundType != "" {
+		return err
+	}
+
+	// Determine if this is a cluster with user defined routing
+	doc.OpenShiftCluster.Properties.NetworkProfile.OutboundType = api.OutboundTypeLoadbalancer
```

Comment:
I think this line should execute in an `else` or default for the `if` immediately after it, rather than always executing and then checking if it's actually correct later.

Reply:
Agreed and updated.